core/src/health.rs

```diff
@@ -0,0 +1,214 @@
+use std::{
+    net::SocketAddr,
+    sync::Arc,
+};
+
+use axum::{
+    extract::State,
+    http::StatusCode,
+    response::Json,
+    routing::get,
+    Router,
+};
+use serde::{Deserialize, Serialize};
+use tokio::net::TcpListener;
+use tracing::{error, info};
+
+use crate::{
+    database::postgres::client::PostgresClient,
+    indexer::task_tracker::active_indexing_count,
+    manifest::core::Manifest,
+    system_state::is_running,
+};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct HealthStatus {
+    pub status: String,
+    pub timestamp: String,
+    pub services: HealthServices,
+    pub indexing: IndexingStatus,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct HealthServices {
+    pub database: String,
+    pub indexing: String,
+    pub sync: String,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct IndexingStatus {
+    pub active_tasks: usize,
+    pub is_running: bool,
+}
+
+#[derive(Clone)]
+pub struct HealthServerState {
+    pub manifest: Arc<Manifest>,
+    pub postgres_client: Option<Arc<PostgresClient>>,
+}
+
+pub struct HealthServer {
+    port: u16,
+    state: HealthServerState,
+}
+
+impl HealthServer {
+    pub fn new(port: u16, manifest: Arc<Manifest>, postgres_client: Option<Arc<PostgresClient>>) -> Self {
+        Self {
+            port,
+            state: HealthServerState {
+                manifest,
+                postgres_client,
+            },
+        }
+    }
+
+    pub async fn start(self) -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
+        let app = Router::new()
+            .route("/health", get(health_handler))
+            .with_state(self.state);
+
+        let addr = SocketAddr::from(([0, 0, 0, 0], self.port));
+        let listener = TcpListener::bind(addr).await?;
+
+        info!("🩺 Health server started on http://0.0.0.0:{}/health", self.port);
+
+        axum::serve(listener, app).await?;
+        Ok(())
+    }
+}
+
+async fn health_handler(State(state): State<HealthServerState>) -> Result<Json<HealthStatus>, StatusCode> {
+    let mut health_status = HealthStatus {
+        status: "healthy".to_string(),
+        timestamp: chrono::Utc::now().to_rfc3339(),
+        services: HealthServices {
+            database: "unknown".to_string(),
+            indexing: "unknown".to_string(),
+            sync: "unknown".to_string(),
+        },
+        indexing: IndexingStatus {
+            active_tasks: active_indexing_count(),
+            is_running: is_running(),
+        },
+    };
+
+    // Check database connection if PostgreSQL is enabled
+    if state.manifest.storage.postgres_enabled() {
+        match &state.postgres_client {
+            Some(client) => {
+                match client.query_one("SELECT 1", &[]).await {
+                    Ok(_) => {
+                        health_status.services.database = "healthy".to_string();
```

Comment:
enums and should be in its own method and make it cleaner we should try not to mut the health object and build it in 1 - same comments for most this file